src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -369,18 +369,14 @@ export function IndexPatternProvider(Private, $http, config, kbnIndex, Promise,
           return confirmModalPromise(confirmMessage, { confirmButtonText: 'Edit existing pattern' })
             .then(() => {
               kbnUrl.change('/management/kibana/indices/{{id}}', { id: duplicate.id });
-              return true;
+              return Promise.reject();
             })
-            .catch(() => {
-              throw new IndexPatternAlreadyExists(this.title);
-            });
+            .catch(() => Promise.reject());
```

Comment:
Unfortunately (and ironically?) the decision to use Promise rejections as the early exit mechanism means that we're once again dealing with error states. This rejection still causes an error state to be triggered, except the error itself is `null`, resulting in a TypeError in `create_index_pattern.js`:

![image](https://user-images.githubusercontent.com/1238659/29234832-adbc54da-7eae-11e7-9aa3-006b63583e7d.png)

The least invasive solution is to probably change that file on line 227, to check for the presence of an error before doing anything with it:

```javascript
      if (err) {
        notify.fatal(err);
      }
```